edge-auth/provider/github_profile.go

```diff
@@ -72,3 +73,56 @@ type GitHubProfile struct {
 	CreatedAt time.Time `json:"created_at"`
 	UpdatedAt time.Time `json:"updated_at"`
 }
+
+// GetUserOrganizations using the API "List your organizations"
+// https://developer.github.com/v3/orgs/#list-your-organizations
+func (gh *GitHub) GetUserOrganizations(accessToken string) (string, error) {
+	apiURL := fmt.Sprintf("https://api.github.com/user/orgs")
+
+	req, reqErr := http.NewRequest(http.MethodGet, apiURL, nil)
+	if reqErr != nil {
+		return "", fmt.Errorf("error while making request to `%s` organizations: %s", apiURL, reqErr.Error())
+	}
+
+	req.Header.Add("Authorization", "token "+accessToken)
+
+	resp, respErr := gh.Client.Do(req)
+	if respErr != nil {
+		return "", fmt.Errorf("error while requesting organizations: %s", respErr.Error())
+	}
+
+	if resp.Body != nil {
+		defer resp.Body.Close()
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("bad status code from request to GitHub organizations: %d", resp.StatusCode)
+	}
+
+	body, bodyErr := ioutil.ReadAll(resp.Body)
+	if bodyErr != nil {
+		return "", fmt.Errorf("error while reading body from GitHub organizations: %s", bodyErr.Error())
+	}
+
+	var orgs []Organization
+	unmarshalErr := json.Unmarshal(body, &orgs)
+	if unmarshalErr != nil {
+		return "", fmt.Errorf("error while un-marshaling organizations: %s, value: %q", unmarshalErr.Error(), body)
+	}
+
+	return formatOrganizations(orgs), nil
+}
+
+func formatOrganizations(orgs []Organization) string {
```

Comment:
This could have a couple of tests

Reply:
Want to find someone in the community to take that?